corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -297,55 +290,38 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
+
             e.target.className += " disabled";
-            this.iconIframe(e, actionUrl, this.model.get('id'));
+            this.clickableIconRequest(e, endpointId, this.model.get('id'), isBackground);
         },
 
-        iconIframe: function (e, url, caseId) {
+        clickableIconRequest: function (e, endpointId, caseId, isBackground) {
             const self = this;
-            const iframeId = "icon-iframe-" + caseId;
             const clickedIcon = e.target;
             clickedIcon.classList.add("disabled");
             clickedIcon.style.display = 'none';
             const spinnerElement = $(clickedIcon).siblings('i');
             spinnerElement[0].style.display = '';
-            const iconIframe = document.createElement('iframe');
-            iconIframe.style.display = 'none';
-            $(iconIframe).attr('id', iframeId);
-            iconIframe.src = encodeURI(url);
-            document.body.appendChild(iconIframe);
-
-            $(`#${iframeId}`).on('load', function () {
-                // Get success or error message from iframe and pass to main window
-                const notificationsElement = $(`#${iframeId}`).contents().find("#cloudcare-notifications");
-                new MutationObserver((el) => {
-                    const addedNodes = el[0].addedNodes;
-                    if (addedNodes[0].classList.contains('alert')) {
-                        const succeeded = addedNodes[0].classList.contains('alert-success');
-                        let message;
-                        if (succeeded) {
-                            message = notificationsElement.find('.alert-success').find('p').text();
-                            FormplayerFrontend.trigger('showSuccess', gettext(message));
-                            self.reloadCase(caseId);
-                        } else {
-                            const messageElement = notificationsElement.find('.alert-danger');
-                            // Todo: standardize structures of success and error alert elements
-                            message = messageElement.contents().filter(function () {
-                                return this.nodeType === Node.TEXT_NODE;
-                            })[0].nodeValue;
-                            FormplayerFrontend.trigger('showError', gettext(message));
-                        }
-                        clickedIcon.classList.remove("disabled");
-                        clickedIcon.style.display = '';
-                        spinnerElement[0].style.display = 'none';
-                        iconIframe.remove();
-                    }
-                }).observe(notificationsElement[0], { childList: true });
+            const currentUrlToObject = formplayerUtils.currentUrlToObject();
+            currentUrlToObject.endpointArgs = {case_id: caseId};
+            currentUrlToObject.endpointId = endpointId;
+            currentUrlToObject.isBackground = isBackground;
```

Comment:
Are we using `isBackground` at the moment to decide on any UI interaction ?

Reply:
It's being used here to display the success message from the `get_endpoint` response. https://github.com/dimagi/commcare-hq/blob/31475c16a6696fd393ce243e3b878140f3cfeb16/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js#L50-L52

Reply:
Think we should always show the message irrespective of  `isBackground`. This flag is more meant to use for whether we want to take user away from the current case list screen based on the response from the clickable icon request. Potentially response from Clickable icon can also be a menu screen that we might wanna take the user to (Think In app links) and in such cases we will have `isBackground` set to `false`. 

Although at the moment it's best to left it unused with a comment since looks like it's not being used meaningfully.